cmd/kaniko-ecr/main.go

```diff
@@ -233,30 +258,37 @@ func run(c *cli.Context) error {
 	return plugin.Exec()
 }
 
-func setupECRAuth(accessKey, secretKey, registry string) error {
-	if registry == "" {
-		return fmt.Errorf("registry must be specified")
+func createDockerConfig(dockerUsername, dockerPassword, accessKey, secretKey, registry string, noPush bool) (*docker.Config, error) {
+	dockerConfig := docker.NewConfig()
+
+	if dockerUsername != "" {
+		dockerConfig.SetAuth(docker.RegistryV1, dockerUsername, dockerPassword)
 	}
 
-	// If IAM role is used, access key & secret key are not required
-	if accessKey != "" && secretKey != "" {
-		err := os.Setenv(accessKeyEnv, accessKey)
-		if err != nil {
-			return errors.Wrap(err, fmt.Sprintf("failed to set %s environment variable", accessKeyEnv))
+	// only setup auth when pushing or credentials are defined
+	if !noPush || accessKey != "" {
+		if registry == "" {
+			return nil, fmt.Errorf("registry must be specified")
 		}
 
-		err = os.Setenv(secretKeyEnv, secretKey)
-		if err != nil {
-			return errors.Wrap(err, fmt.Sprintf("failed to set %s environment variable", secretKeyEnv))
+		// If IAM role is used, access key & secret key are not required
+		if accessKey != "" && secretKey != "" {
+			err := os.Setenv(accessKeyEnv, accessKey)
+			if err != nil {
+				return nil, errors.Wrap(err, fmt.Sprintf("failed to set %s environment variable", accessKeyEnv))
+			}
+
+			err = os.Setenv(secretKeyEnv, secretKey)
+			if err != nil {
+				return nil, errors.Wrap(err, fmt.Sprintf("failed to set %s environment variable", secretKeyEnv))
+			}
 		}
-	}
 
-	jsonBytes := []byte(fmt.Sprintf(`{"credStore": "ecr-login", "credHelpers": {"%s": "ecr-login", "%s": "ecr-login"}}`, ecrPublicDomain, registry))
-	err := ioutil.WriteFile(dockerConfigPath, jsonBytes, 0644)
-	if err != nil {
-		return errors.Wrap(err, "failed to create docker config file")
+		dockerConfig.SetCredHelper(ecrPublicDomain, "ecr-login")
```

Comment:
Any reason not to set `"credStore": "ecr-login"`? It is present in kaniko doc https://github.com/GoogleContainerTools/kaniko#pushing-to-amazon-ecr